File: Algorithms/Sorting-Algorithms/selection_sort.py
"""Demo code for selection sort
"""


def selectionSort(array: list) -> list:
    """Sorts an list in ascending order using selection sort algorithm

    Args:
        array (list): The unsorted list

    Returns:
        list: The sorted list
    """
    n = len(array)
    for i in range(n):
        min_val = array[i]
        t = i
        for j in range(i, n):
            if array[j] < min_val:
                min_val = array[j]
                t = j
        array[i], array[t] = array[t], array[i]
    return array


if __name__ == "__main__":
    data = [10, 5, 2, 4, 8]
    print(selectionSort(data))

File: Algorithms/Sorting-Algorithms/Selection-Sort/selection_sort.js
// JavaScript code for selection sort

/**
 * Sorts an array of numbers in ascending order using the selection sort algorithm
 * @param {array} data The unsorted array
 * @returns The sorted array
 */
function selection_sort(data){
    let n = data.length;
    for (let i = 0; i < n; i++){
        let min_val = data[i]; // The minimum value after index `i`
        let t = i  // The index of the minimum value
        for (let j = i; j < n; j++){
            if (data[j] < min_val){
                min_val = data[j];
                t = j;
            }
        }
        // Swap the ith value with the smallest value after it
        let temp = data[i];
        data[i] = data[t];
        data[t] = temp;
    }
    return data
}


const DATA = [10, 5, 2, 4, 8];
console.log(selection_sort(DATA));
